templates/ovndbcluster/bin/setup.sh

```diff
@@ -73,4 +79,44 @@ set "$@" --ovn-${DB_TYPE}-log=-vconsole:{{ .OVN_LOG_LEVEL }}
 set "$@" --ovn-${DB_TYPE}-logfile=/dev/null
 
 # don't log to file (we already log to console)
-$@ ${OPTS} run_${DB_TYPE}_ovsdb -- -vfile:off
+$@ ${OPTS} run_${DB_TYPE}_ovsdb -- -vfile:off &
+
+# Once the database is running, we will attempt to configure db options
+CTLCMD="ovn-${DB_TYPE}ctl --no-leader-only"
+
+# TODO: this logic is dedicated for the leader, not necessarily the -0 pod. The
```

Comment:
The reasoning here was to run the db commands only on 1 of the node and considering -0 will be always there when have replicas>=1 it was choosen.

Reply:
Ack, will update the comment. Curious: does it mean that we could do that in every pod with the same effect?

Also, doesn't `--no-leader-only` mean that, potentially, the remote will NOT be configured if pod0 is not leader when a config change for ssl or inactivity-probe was applied? (Will non-leader propagate the change to leader?)

Reply:
Doing it in every pod will just duplicate the requests as write requests are propagated to leader as described in the doc https://github.com/ovn-org/ovn/blob/cb5d5bb3d8d2bed131112c42cab0d2ba5af8e53f/utilities/ovn-nbctl.8.xml#L213

Reply:
@karelyatin but if pod-0 is not the leader, and it runs this setup as --no-leader-only, then there is a chance, I think, that this update will not get to the leader? just thinking if perhaps only the leader (whoever it is) should execute these commands, not necessarily pod-0. Or do we think that non-leader changes will always eventually propagate to leaders?

(This would not be a big deal if commands affected bootstrap only, but I believe configuration may change and so commands have to be applied on each startup.)

Reply:
@booxter /me not sure about the case when the request not forwarded for the leader(may be you referring some network issue?), if it can't connect the leader i expect those db commands to fail and stop. And regarding the configuration change post bootstrap is a valid case.

Reply:
@karelyatin do we assume that these commands will fail if non-leader pod-0 can't communicate to the leader? I am not so sure. I think we start our members independently, and we don't require for a pod to be alive to communicate with the leader. So I can imagine, in some cases, a scenario where pod-0 (non-leader) is started; these ctl commands are executed on pod-0; since --no-leader-only is passed, the commands communicate with the local member (pod-0), not the leader; they succeed; pod-0 attempts to propagate the update to the leader (?) but fails - network issue, or maybe some raft protocol hiccup; maybe pod-0 got detached from the cluster somehow... long story short, it can't sync inactivity probe into cluster; then the rest of pods in the cluster continue running without knowing about the new inactivity probe value. (The probe value here is just an example; other ctl commands we run with --no-leader-only are affected I think the same way.)

I am thinking that perhaps it would be safer for all members to run these commands against itself? Is there some side effect of doing it that is not desirable?

Reply:
<< these ctl commands are executed on pod-0; since --no-leader-only is passed, the commands communicate with the local member (pod-0), not the leader; they succeed; pod-0 attempts to propagate the update to the leader (?) but fails - network issue, or maybe some raft protocol hiccup; maybe pod-0 got detached from the cluster somehow
@booxter from what i understood reading the document shared above, that for write/dbmodify operations it will forward the request to leader even if we pass --no-leader-only i.e and it will not apply locally and shouldn't succeed, i need to simulate this to confirm the behaviour. we can deal with this in follow up, for now let's merge it.
For point regarding running on all nodes I am not aware about side-effects of running the commands on all nodes.